cfme/utils/ssh.py

```diff
@@ -204,9 +203,11 @@ def _progress_callback(self, filename, size, sent):
             logger.debug('scp progress for %r: %s of %s ', filename, sent, size)
 
     def close(self):
-        with diaper:
-            _client_session.remove(self)
         super(SSHClient, self).close()
+        try:
+            _client_session.remove(self)
+        except (AttributeError, ValueError):
```

Comment:
Q: Why not log this exception here?

Reply:
I literally couldn't get logger to work in this context, I believe because its in GC when this is happening. Any calls to logger method resulted in those internal logger functions raising TypeErrors.